jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/QueryServiceImplTest.java

```diff
@@ -443,6 +451,53 @@ public void testGetTaskInstancesAsPotOwners() {
     }
 
     @Test
+    public void testGetTaskInstancesAsExcludedOwner() {
+        query = new SqlQueryDefinition("getMyTaskInstances", dataSourceJNDIname, Target.PO_TASK);
+        query.setExpression(PO_TASK_QUERY);
+
+        queryService.registerQuery(query);
+
+        List<QueryDefinition> queries = queryService.getQueries(new QueryContext());
+        assertNotNull(queries);
+        assertEquals(1, queries.size());
+
+        QueryDefinition registeredQuery = queries.get(0);
+        assertNotNull(registeredQuery);
+        assertEquals(query.getName(), registeredQuery.getName());
+        assertEquals(query.getSource(), registeredQuery.getSource());
+        assertEquals(query.getExpression(), registeredQuery.getExpression());
+        assertEquals(query.getTarget(), registeredQuery.getTarget());
+
+        registeredQuery = queryService.getQuery(query.getName());
+
+        assertNotNull(registeredQuery);
+        assertEquals(query.getName(), registeredQuery.getName());
+        assertEquals(query.getSource(), registeredQuery.getSource());
+        assertEquals(query.getExpression(), registeredQuery.getExpression());
+        assertEquals(query.getTarget(), registeredQuery.getTarget());
+
+        processInstanceId = processService.startProcess(deploymentUnit.getIdentifier(), "org.jbpm.ExcludedOwner", new HashMap<String, Object>());
+        assertNotNull(processInstanceId);
+
+        identityProvider.setName("maciej");
+        identityProvider.setRoles(Arrays.asList("admins"));
+
+        List<UserTaskInstanceDesc> taskInstanceLogs = queryService.query(query.getName(), UserTaskInstanceQueryMapper.get(), new QueryContext());
+        assertNotNull(taskInstanceLogs);
+        assertEquals(1, taskInstanceLogs.size());
+
+        identityProvider.setName("kris");
+        identityProvider.setRoles(Arrays.asList("admins"));
+
+        taskInstanceLogs = queryService.query(query.getName(), UserTaskInstanceQueryMapper.get(), new QueryContext());
+        assertNotNull(taskInstanceLogs);
+        assertEquals(0, taskInstanceLogs.size());
+
+        processService.abortProcessInstance(processInstanceId);
+        processInstanceId = null;
```

Comment:
You don't have to clean up the reference at the test end, JUnit instantiates a new class for each ```@Test``` method.
If you want to keep consistency with the rest of the tests in the class, I guess you can leave it as it is but,just so you know the statement is redundant.

Reply:
@bsremac it seems that _processInstanceId_ is used in the _cleanup_ method to abort the process instance in case is not null already.

Reply:
I guess in this case you could relax the test from any cleanup steps after the last assertion and let the cleanup method do it's job :)